lib/view_component/compile_cache.rb

```diff
@@ -20,10 +20,18 @@ def compiled?(klass)
 
     def invalidate_class!(klass)
       cache.delete(klass)
+
+      klass.send(:undef_method, :render_template_for)
```

Comment:
By using remove_method instead of undef_method we can remove the method from the subclass letting the version defined in Base be again the one that is being called. That should save some time in dev/test, but most importantly simplify. 

Maybe worth at that point to ensure we're never redefining or removing it from Base.

Reply:
Wow, I didn't know there was a difference between `remove_method` and `undef_method`! Even after 12 years, Ruby can still surprise me ❤️

Reply:
Come on over to [Opal](https://github.com/opal/opal/blob/6f84b76717c2c56e031baa40d3309c1562405577/opal/corelib/module.rb#L410-L418) and make some ruby specs pass, there's plenty of them that keep surprising me! 😅😊